src/_components/AssetDetails/AssetDetailsContainer.jsx

```diff
@@ -1,87 +1,23 @@
-import React, { Component } from 'react';
 import { connect } from 'react-redux';
-import PropTypes from 'prop-types';
-import { isEmpty } from 'lodash';
-import { Container, Divider, Header } from 'semantic-ui-react';
+import { get, isEmpty } from 'lodash';
+
 import { getAssetDetail } from '../../_actions/asset.actions';
 import { loadAssetAssigneeUsers } from '../../_actions/users.actions';
 import AssetDetailsComponent from '../../components/AssetDetails/AssetDetailsComponent';
-import NavBarComponent from '../NavBarContainer';
-import LoaderComponent from '../../components/LoaderComponent';
-
-export class AssetDetailsContainer extends Component {
-  state = {
-    assignedUser: {},
-    hasError: this.props.hasError,
-    errorMessage: this.props.errorMessage
-  };
 
-  componentDidMount() {
-    const { match } = this.props;
-    const { id } = match.params;
+// Ensure you are comparing the correct ids. I have used `assetDetail.id` for demonstration purpose
+export const isSameId = (assetDetail, props) => (assetDetail.id === props.match.params.id);
 
-    this.props.getAssetDetail(id);
-    this.props.loadAssetAssigneeUsers();
+// If same id, then the data exist in store. Otherwise, check for it in props.location.state
+export const fetchAssetDetail = (assetDetail, props) => {
```

Comment:
I think `getAssetInfo` makes more sense as `fetchAssetDetail` sounds more like we are making an API request, which is not the case